components/user_settings/general/user_settings_general.jsx

```diff
@@ -118,13 +118,11 @@ class UserSettingsGeneralTab extends React.Component {
         this.submitActive = false;
 
         this.state = this.setupInitialState(props);
-    }
 
-    componentWillMount() {
-        const verificationEnabled = this.props.sendEmailNotifications && this.props.requireEmailVerification;
+        const verificationEnabled = props.sendEmailNotifications && props.requireEmailVerification;
 
-        if (verificationEnabled && !this.props.user.email_verified) {
-            this.setState({emailChangeInProgress: true});
+        if (verificationEnabled && !props.user.email_verified) {
+            this.state = Object.assign({}, this.state, {emailChangeInProgress: true});
```

Comment:
Sorry to be nit-picky (and my fault for not noticing this in the initial review) but this would probably fit better in the `setupInitialState` function that prepares the rest of the state in the constructor

Reply:
Yeah I was thinking about that but that function (`setUpInitialState`) is actually called a few other times in that component and I didn't want to make that check happen every time the function is called since it should only happen once when the component renders. What do you think?

Reply:
For example here:
```
    updateSection = (section) => {
        const emailChangeInProgress = this.state.emailChangeInProgress;
        this.setState(Object.assign({}, this.setupInitialState(this.props), {emailChangeInProgress, clientError: '', serverError: '', emailError: '', sectionIsSaving: false}));
        this.submitActive = false;
        this.props.updateSection(section);
    }
```
I can just update this to remove `emailChangeInProgress` since it is handled in setUpInitialState?